cms/server/admin/handlers/base.py

```diff
@@ -410,6 +411,40 @@ def get_score_type(self, dest, name, params):
         dest["score_type"] = name
         dest["score_type_parameters"] = params
 
+    def render_params_for_submissions(self, query, page, page_size=50):
+        """Add data about the requested submissions to r_params.
+
+        submission_query (sqlalchemy.orm.query.Query): the query
+            giving back all interesting submissions.
+        page (int): the index of the page to display.
+        page_size(int): the number of submissions per page.
+
+        """
+        query = query\
+            .options(joinedload(Submission.task))\
+            .options(joinedload(Submission.participation))\
+            .options(joinedload(Submission.files))\
+            .options(joinedload(Submission.token))\
+            .options(joinedload(Submission.results))\
+            .order_by(Submission.timestamp.desc())
```

Comment:
Some of the code that uses this function will then access, for example, the SubmissionResult's evaluations. Maybe we should join-load these here as well, to avoid a query being fired for each submission in those cases. (For example, the contestsubmission handler shows a table with all evaluations in case the submission is still scoring)